### ISSUE TITLE
tests/end_to_end/test_gestalt should use SimulatedBackend.

### ISSUE THREAD

**Issue by jeromekelleher** [ga4gh/server#226]
The end-to-end tests should allocate a SimulatedBackend, and run some simple tests on it. For example, we should

1. Write a server config file to a tmp file, and pass this to the server using the '-f' argument. This allows us to set the simulator parameters, such as VARIANT_DENSITY and NUM_VARIANT_SETS.
2. We can then test the output of the client to see if it matches. For example, we can run a `search-variantsets`, and verify the number of lines written to stdout is correct. If we set VARIANT_DENSITY to 1, then we can also verify that the number of lines written to stdout for a variants-search is correct. For example, if we run
```
$ python client_dev.py variants-search http://localhost:8000/v0.5.1 -s0 -e2 
```
we should get 2*NUM_VARIANT_SETS lines in the output. This tests a lot of different aspects of the system and would be a very useful test.

**Comment by dcolligan** [comment]
Running that client command doesn't produce any results.

With:
```
$ python server_dev.py -c TestConfig
```
We get:
```
$ python client_dev.py -vv variants-search http://localhost:8000/v0.5.1 -s0 -e2
ga4gh.client: POST http://localhost:8000/v0.5.1/variantsets/search
ga4gh.client: json request:
ga4gh.client: {
    "datasetIds": [],
    "pageSize": null,
    "pageToken": null
}
ga4gh.client: json response:
ga4gh.client: {
    "nextPageToken": null,
    "variantSets": []
}
ga4gh.client: POST http://localhost:8000/v0.5.1/variants/search
ga4gh.client: json request:
ga4gh.client: {
    "callSetIds": [],
    "end": 2,
    "pageSize": 100,
    "pageToken": null,
    "referenceName": "1",
    "start": 0,
    "variantName": null,
    "variantSetIds": []
}
ga4gh.client: json response:
ga4gh.client: {
    "nextPageToken": null,
    "variants": []
}
```

**Comment by jeromekelleher** [comment]
We can't just use the bottled `TestConfig` here @dcolligan because it has an '__EMPTY__' variant set. We do need to set the config explicitly, or we can't test that the returned values are correct.

**Comment by dcolligan** [comment]
Ah, forgot about the distinction between empty and simulated.

**Comment by dcolligan** [comment]
Fixed in #230
